src/node/util.ts

```diff
@@ -115,10 +118,162 @@ export const generatePassword = async (length = 24): Promise<string> => {
   return buffer.toString("hex").substring(0, length)
 }
 
-export const hash = (str: string): string => {
+/**
+ * Used to hash the password.
+ */
+export const hash = async (password: string): Promise<string> => {
+  try {
+    return await argon2.hash(password)
+  } catch (error) {
+    logger.error(error)
+    return ""
+  }
+}
+
+/**
+ * Used to verify if the password matches the hash
+ */
+export const isHashMatch = async (password: string, hash: string) => {
+  try {
+    return await argon2.verify(hash, password)
```

Comment:
:warning: `password` has gotten quite deep without any actual validation of its shape. It’s was type asserted a few calls up. It could be an object with a prototype pollution if the Express handler didn’t perform any sanitization of that.

Reply:
That's a really good point! Thanks for flagging that. I guess I probably want to write a function that sanitizes the password somewhere before we get to this point. I'll look into this (suggestions welcome too).

Reply:
I added some basic sanitization up the chain for when we use `req.body.password` and `req.cookie.key` which should address this.